File: exercises/python_playwright/Mouse_Over/mouse_over_playwright.py
from playwright.sync_api import Page, expect


class MouseOverPage:
    def __init__(self, page: Page):
        """Page object for the Mouse Over exercise."""
        self.page = page
        self.click_me_btn = page.get_by_title("Click Me")
        self.click_me_btn_active = page.get_by_title("Active Link")
        self.link_btn = page.get_by_title("Link Button")
        self.click_me_click_count = page.locator("#clickCount")
        self.link_btn_click_count = page.locator("#clickButtonCount")

    def open(self) -> None:
        """Open the Mouse Over exercise page."""
        self.page.goto("http://uitestingplayground.com/mouseover")

    def doubleclick_button_click_me(self) -> None:
        """Double-click the Click Me control after hovering it."""
        self.click_me_btn.hover()
        expect(self.click_me_btn_active).to_have_class("text-warning")
        self.click_me_btn_active.dblclick()

    def doubleclick_button_link_button(self) -> None:
        """Double-click the Link Button after hovering it."""
        self.link_btn.hover()
        expect(self.link_btn).to_have_class("text-warning")
        self.link_btn.dblclick()

File: exercises/javascript_playwright/mouse-over/mouseOverPlaywright.js
export class MouseOverPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.clickMeLink = page.getByTitle("Click me");
        this.clickMeLinkActive = page.getByTitle("Active Link");
        this.linkButton = page.getByTitle("Link Button");
        this.clickMeCounter = page.locator("#clickCount");
        this.linkButtonCounter = page.locator("#clickButtonCount");
    }

    async openExercisePage() {
        await this.page.goto("http://uitestingplayground.com/mouseover");
    }

    async hoverClickMeLink() {
        await this.clickMeLink.hover();
    }

    async hoverLinkButton() {
        await this.linkButton.hover();
    }

    async doubleClickClickMeLink() {
        await this.clickMeLinkActive.dblclick();
    }

    async doubleClickLinkButton() {
        await this.linkButton.dblclick();
    }
}
